tests/models/test_selector.py

```diff
@@ -163,14 +158,10 @@ def test_models_selector_model_selector_model_rules(model_rules, rules):
         model_rules[BaseModel] = rules
 
 
-@pytest.mark.parametrize(
-    "event,model",
-    [
-        (ServerEventFactory(), ServerEventModel),
-        (PageCloseBrowserEventFactory(), PageCloseBrowserEventModel),
-    ],
-)
-def test_models_selector_model_selector_get_model_with_valid_event(event, model):
+@pytest.mark.parametrize("model", [PageClose])
+@settings(max_examples=1)
+@given(st.builds(PageClose, referer=provisional.urls(), page=provisional.urls()))
+def test_models_selector_model_selector_get_model_with_valid_event(model, event):
```

Comment:
Is it possible to include the `ServerEvent` case in this test too ? If not, maybe we could split this test in two tests - one for `ServerEvent` and one for `PageClose` ?

Reply:
Why do you want to test this mechanism with two examples? Once it has been tested with one event, I think it is not necessary to test with PageClose? WDYT?

Reply:
The Idea is to use this test for all event models we define. It assures that for a given event the `get_model` method returns the corresponding model.

Reply:
I think that is a point we need to discuss IRL. I would not implement this test for two reasons: 
1. we test the selector mechanism here and this mechanism is not event-specific
2. maintain the tests for all the event would be very huge and maybe too hard to maintain when having all events implemented
WDYT ?

Reply:
Ok, let's discuss this IRL) Indeed this test might get huge with a lot of events, you are right. But for me this mechanism is event-specific as each event implements the `__selector__` field which is responsible for the model retrieval in `get_model`.

Reply:
As discussed with @quitterie-lcs, model selectors can be tested in the model's test suite, not here. Only the model selector logic may be tested here. WDYT?

Reply:
@SergioSim I got the point of testing `get_model` for each event. Hence, we can move both tests `test_models_selector_model_selector_get_model_with_valid_event` and `test_models_selector_model_selector_get_model_with_invalid_event` in the corresponding model test file?

Reply:
Good point) Yes, this would be a solution) If I understood it right, we would write for each model a test case with the ModelSelector in the corresponding model's test suite ? I could propose to automatize this test in a single one:

1. for each `ModelSelector` supported model:
2. generate `test_event` with hypothesis
3. check that `ModelSelector.get_model(test_event)` returns the corresponding model.

WDYT ?

Reply:
It's a security check that can be automated indeed, but I'm not sure it's really valuable. I would like to battle test our selectors on a model basis in various conditions.